graylog2-server/src/main/java/org/graylog/events/processor/aggregation/AggregationEventProcessor.java

```diff
@@ -219,7 +220,7 @@ private Set<String> getStreams(AggregationEventProcessorParameters parameters) {
 
     private void filterSearch(EventFactory eventFactory, AggregationEventProcessorParameters parameters,
                               EventConsumer<List<EventWithContext>> eventsConsumer) throws EventProcessorException {
-        final Set<String> streams = getStreams(parameters);
+        final Set<String> streams = eventStreamService.buildEventSourceStreams(getStreams(parameters), Collections.emptySet());
```

Comment:
I think this is the wrong method, because, as the name suggests, it is supposed to operate on the results of a search. I think we should use the same method that we use for aggregations [here](https://github.com/Graylog2/graylog2-server/blob/bf91ef97a613a20df8d37aba7879445d8ff5ba45/graylog2-server/src/main/java/org/graylog/events/processor/aggregation/PivotAggregationSearch.java#L372).

It appears as if that one will also take care of filtering out the investigation streams, if the enterprise plugin is present. Otherwise we'll end up using different stream filters on the search queries for aggregation and filter events.